src/modules/socket/server/raw_socket.cpp

```diff
@@ -16,6 +16,17 @@ namespace openperf {
 namespace socket {
 namespace server {
 
+/* The raw_socket class casts from dgram_ip_addr to ip_addr_t so both
+ * structures must have exactly the same memory layout
+ */
+static_assert(sizeof(struct dgram_ipv4_addr) == sizeof(ip4_addr_t));
+static_assert(sizeof(struct dgram_ipv6_addr) == sizeof(ip6_addr_t));
+static_assert(sizeof(((struct dgram_ip_addr*)0)->type)
+              == sizeof(((ip_addr_t*)0)->type));
+static_assert(offsetof(struct dgram_ip_addr, type)
+              == offsetof(struct ip_addr, type));
+static_assert(sizeof(struct dgram_ip_addr) == sizeof(ip_addr_t));
+
```

Comment:
Added a bunch of static_asserts() so we can tell if the strucutres don't match up again.